### DISCUSSION TITLE
Silly question : can the autoscaler be used on instances which are not spot ones ?

### DISCUSSION THREAD

**Discussion by ItsTimeToWakeUp** [jcjorel/clonesquad-ec2-pet-autoscaler#14] [General]
Hi, 
I read the doc but maybe with not being focus enough cause I think I missed an info.
I see you're talking about Spot instance credit cpu usage when dealing with autoscaling.

In our application, we don't use such instances. Only 'pet' on demand instances.

I'd like to know if your autoscalling can be applied in theses conditions ? For example, I have a production env which a third is swittched off during the night thanks to clonesquad. Can the same api , with the good cloud watch alarms, do the job in case of failure of the other prod ?
My usecase "the prod which is active just broke , an automatic mecanism make the traffic to route to the non active prod which only one third is on. Not enough for the new load. Can the autsoscaller make the sleepy prod getting awake ?"

Sorry for my english and thx for the project  !
We use it here :)

Greetings

**Comment by jcjorel** [comment]
Hi,
CloneSquad supports but not requires Spot instances: You can use regular on-demand instances.

Regarding your need to start other instances on failure, if you are using min_instance_count ou desired_instance_count specifying a percentage of 33% for example, CloneSquad will start automatically other instances to keep the target percentage of “sane” instances. What matters here is how CloneSquad can detect if instances are healthy: CS will use standard health checks of AWS LoadBalancers if your instances are part of the associated target group. 
If you do not use target group, indeed, you need to define a CloudWatch alarm to ask CS to scale the fleet. But, as it is less integrated in the AWS ecosystem, you should try to simply use target group health checks to do the job…

For info, you can easily test the replacement behavior by simply shutdowning one of your running instances; CS should restart quickly another instance to keep the target pourcentage target…

And, more generally, remember that CS is to help Lift&Shift migrations, and even if CS is currently used in large Production env, if you are designing a new application, consider using AWS AutoScaling instead 😉

**Reply by ItsTimeToWakeUp** [reply]
Thank you for the reply ! We are totally in the case of Lift&Shift migration :) And your product is used by all the company ;)
Thks for the job !
